sled-agent/src/sled_agent.rs

```diff
@@ -360,15 +360,21 @@ impl SledAgent {
 
         // Begin monitoring the underlying hardware, and reacting to changes.
         let sa = sled_agent.clone();
+        let hardware_log = log.clone();
         tokio::spawn(async move {
-            sa.hardware_monitor_task(log).await;
+            sa.hardware_monitor_task(hardware_log).await;
         });
 
         // Finally, load services for which we're already responsible.
         //
         // Do this *after* monitoring for harware, to enable the switch zone to
         // establish an underlay address before proceeding.
         sled_agent.inner.services.load_services().await?;
+
+        // Now that we've initialized the sled services, notify nexus again
+        // at which point it'll plumb any necessary firewall rules back to us.
+        sled_agent.notify_nexus_about_self(&log);
```

Comment:
Note there's already a call to `notify_nexus_about_self` just above but since that's before we've actually initialized any services a Nexus running on a different sled could send us the right rules but we'd have no OPTE ports created yet to apply them to.

Reply:
It makes 100% sense to me *why* we're doing this, but it feels a little odd.

The fact that we store all the services necessary for executing (simply to bootstrap the system) makes sense, so the sled can launch them locally, but all firewall rules are deleted between reboots feels... quirky? Doesn't this mean that in the period of time between "services have started" and "sled agent has notified Nexus, re-received the firewall rules, and re-applied them" we're basically ignoring all user-requested firewall rules? Is that a concern?

Reply:
These are firewall rules for services running on a sled (nexus & external dns in particular) and not guest instances. So yea, there is a bit of a window between those services coming up and actually being externally accessible.

In theory it's possible an instance could be provisioned on the sled in that window and that would be fine as any firewall rules it needs are plumbed along with it during its initialization on the sled.